test/forms/NumberOfPackagesFormProviderSpec.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import forms.behaviours.FieldBehaviours
+import generators.Generators
+import models.{Index, NormalMode}
+import models.messages.UnloadingRemarksRequest
+import org.scalacheck.Gen
+import play.api.Application
+import play.api.data.{Field, Form, FormError}
+import play.api.i18n.Lang.defaultLang
+import play.api.i18n.{Messages, MessagesApi, MessagesImpl}
+import play.api.inject.guice.GuiceApplicationBuilder
+import viewModels.houseConsignment.index.items.NumberOfPackagesViewModel
+
+import scala.util.matching.Regex
+
+class NumberOfPackagesFormProviderSpec extends FieldBehaviours with Generators {
+
+  private val maxLength  = UnloadingRemarksRequest.numberOfPackagesLength
+  private val invalidKey = "numberOfPackages.error.nonNumeric"
+
+  private val viewModel = NumberOfPackagesViewModel(hcIndex, itemIndex, NormalMode)
+
```

Comment:
Possibly do checkmode as well? the mandatory field spec would be different:
 behave like mandatoryField(
        form,
        fieldName,
        requiredError = FormError(fieldName, messages(requiredKey, hcIndex.display, itemIndex.display))
      )